### PR TITLE
NEW SearchableDropdownField

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1618

This code is adapted from [silverstripe/tagfield](https://github.com/silverstripe/silverstripe-tagfield/)

Framework PR
- https://github.com/silverstripe/silverstripe-framework/pull/11057

### CODE REVIEW

**Comment by emteknetnz** [comment]
Design feedback:
- try and remove arrow for lazy load
- get rid of mid-grey hover state
- get rid of 'active hover' state blue, make it mid-blue 'active non-hover state'

- change padding on multi-selected options - react-select-9jq23d 
padding: 3px 10px 3px 10px;

.ss-searchable-dropdown-field__multi-value__remove
padding: 2px 5px 2px 5px;

.ss-searchable-dropdown-field__multi-value
change border color from #66afe9 (light blue) to grey #c1cad9

.ss-searchable-dropdown-field__multi-value
border-left: #c1cad9

.ss-searchable-dropdown-field__multi-value__remove:focus, .ss-searchable-dropdown-field__multi-value__remove:hover
background-color: #ffe0e0 (randomly grabbed, ideally would have an existing variable, or use calc() on $branch-danger)
color: #d40404 ($brand-danger)

**Comment by emteknetnz** [comment]
Design review implementation has been approved by internal design team

File: client/src/bundles/bundle.js [hunk 1]
```diff
@@ -81,6 +81,7 @@ import 'expose-loader?exposes=formatWrittenNumber!lib/formatWrittenNumber';
 import 'expose-loader?exposes=withDragDropContext!lib/withDragDropContext';
 import 'expose-loader?exposes=withRouter!lib/withRouter';
 import 'expose-loader?exposes=ssUrlLib!lib/urls';
+import 'expose-loader?exposes=SearchableDropdownField!components/SearchableDropdownField/SearchableDropdownField';
```
Comment (ChloeSartorelli) [inline client/src/bundles/bundle.js:84]: Exposing things through here needs to be coupled with a PR to the webpack config, or people can't use it.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Code looks broadly good.
Haven't fully tested it locally yet - will wait for the framework changes.

One thing I did notice: when I type to search, it swaps the dirty state of the form, making it look like there are unsaved changes.

IIRC we had that problem with tagfield somewhat recently as well.

File: client/src/bundles/bundle.js (hunk 1 above)
Comment (emteknetnz) [reply client/src/bundles/bundle.js:84]: https://github.com/silverstripe/webpack-config/pull/64

**Comment by emteknetnz** [comment]
I think I've fixed the issue with the fix the dirty state on search

**Comment by GuySartorelli** [comment]
If I select an option and then clear that option (without saving inbetween), the page still thinks I have unsaved changes.

1. Page loads. No unsaved changes.
2. I type a search term. No unsaved changes.
3. I select an option. Page correctly identifies changes.
4. I clear the selection. Page incorrectly thinks there are still unsaved changes.

**Comment by emteknetnz** [comment]
I've tried really hard to fix this though I just cannot.  There's a 'changed' class that gets added by [jquery.changetracker](https://github.com/silverstripe/silverstripe-admin/blob/2/client/src/legacy/jquery.changetracker.js#L63) to an `<input>` that react-select added.  `jquery.changetracker` just blindly tracks virtually all inputs, which is great until you get situations like this.

The issue seems be that `jquery.changetracker` has full control over the 'changed' class being added and there's nothing you can do about it.

Setting a `no-change-track` className on the `<DynamicComponent>` will block all change tracking.  I've tried that along with also adding in a new `<input>` property as a sibling to `<DynamicComponent>`, manually setting the className on the input to 'changed' based on whether there are changes and dispatching the change event from that element. But that doesn't work, I still don't get to control whether or not the 'changed' class appears on the input

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Please add behat tests - especially for https://github.com/silverstripe/silverstripe-framework/pull/11057#discussion_r1424715991

**Comment by emteknetnz** [comment]
We don't have behat tests for FormFields in admin, instead we use jest tests.  This PR has jest tests.  For the 'clearable' functionality called out in https://github.com/silverstripe/silverstripe-framework/pull/11057#discussion_r1424715991 - that's functionality on [react-select](https://react-select.com) so all the PHP code is doing is passing a prop to it

**Review by GuySartorelli — APPROVED** [review]
Looks good, works great locally.

**Pull request merged**